lib/signs/utilities/audio.ex

```diff
@@ -11,6 +11,108 @@ defmodule Signs.Utilities.Audio do
+  def from_sign(sign) do
+    {Enum.flat_map(decode_sign(sign), &get_passive_readout/1), sign}
+  end
+
+  @spec get_passive_readout(
+          {atom(), Content.Message.t(), Content.Message.t() | nil}
+          | {atom(), [Content.Message.t()]}
+        ) :: [Content.Audio.t()]
+  defp get_passive_readout({:custom, top, bottom}) do
+    Audio.Custom.from_messages(top, bottom)
+  end
+
+  defp get_passive_readout({:headway, top, bottom}) do
+    case top do
+      %Message.Headways.Top{} ->
+        Audio.VehiclesToDestination.from_headway_message(top, bottom)
+
+      %Message.Headways.Paging{} ->
+        Audio.VehiclesToDestination.from_paging_headway_message(top)
+    end
+  end
+
+  defp get_passive_readout({:scheduled_train, top, bottom}) do
+    case top do
+      %Message.EarlyAm.DestinationTrain{} ->
+        Audio.FirstTrainScheduled.from_messages(top, bottom)
+
+      %Message.EarlyAm.DestinationScheduledTime{} ->
+        Audio.FirstTrainScheduled.from_messages(top)
+    end
+  end
+
+  defp get_passive_readout({:alert, top, bottom}) do
+    case top do
+      %Message.Alert.NoService{} ->
+        Audio.Closure.from_messages(top, bottom)
+
+      %Message.Alert.DestinationNoService{} ->
+        Audio.NoServiceToDestination.from_message(top)
+
+      %Message.Alert.NoServiceUseShuttle{} ->
+        Audio.NoServiceToDestination.from_message(top)
+    end
+  end
+
+  defp get_passive_readout({:predictions, predictions}) do
+    case predictions do
+      [
+        %Message.StoppedTrain{destination: same} = top,
+        %Message.StoppedTrain{destination: same}
+      ] ->
+        Audio.StoppedTrain.from_message(top)
+
+      [
+        %Message.StoppedTrain{destination: same} = top,
+        %Message.Predictions{destination: same} = bottom
+      ] ->
+        Audio.StoppedTrain.from_message(top) ++
+          Audio.FollowingTrain.from_predictions_message(bottom)
+
+      [
+        %Message.Predictions{destination: same} = top,
+        %Message.StoppedTrain{destination: same}
+      ] ->
+        get_prediction_readout(top)
+
+      [
+        %Message.Predictions{destination: same} = top,
+        %Message.Predictions{destination: same} = bottom
+      ] ->
+        get_prediction_readout(top) ++
+          Audio.FollowingTrain.from_predictions_message(bottom)
+
+      _ ->
+        Enum.flat_map(predictions, &get_prediction_readout/1)
+    end
+  end
+
+  defp get_prediction_readout(%Message.Predictions{minutes: minutes} = prediction) do
+    case minutes do
+      :boarding ->
+        Audio.TrainIsBoarding.from_message(prediction)
+
+      :arriving ->
+        Audio.TrainIsArriving.from_message(prediction)
+
+      :approaching ->
+        Audio.NextTrainCountdown.from_message(%{prediction | minutes: 1})
+
+      minutes when is_integer(minutes) ->
+        Audio.NextTrainCountdown.from_message(prediction)
+
+      _ ->
+        []
+    end
+  end
+
+  defp get_prediction_readout({%Message.StoppedTrain{} = prediction, _}) do
```

Comment:
This signature looks incorrect, should probably be:
```suggestion
  defp get_prediction_readout(%Message.StoppedTrain{} = prediction) do
```

Reply:
woops yes, good catch